lib/src/facts/posix/uptime_resolver.cc

```diff
@@ -43,6 +48,27 @@ namespace facter { namespace facts { namespace posix {
 
     int64_t uptime_resolver::get_uptime()
     {
+        // We first try to get the system boot time by reading it directly from
+        // the utmp file using the getutxtent system call. This approach
+        // was adapted from https://stackoverflow.com/questions/2071689/how-can-i-get-the-uptime-of-a-ibm-aix-box-in-seconds?answertab=votes#tab-top
+        LOG_DEBUG("Attempting to calculate the uptime from the utmp file...");
+        setutxent();
+        int64_t boot_time = -1;
+        struct utmpx* entry;
+        while ((entry = getutxent())) {
+          if (!strcmp("system boot", entry->ut_line)) {
```

Comment:
I also tested that we correctly delegate to `uptime` by changing this entry to something nonsensical (e.g. "system foo") if we cannot find the `boot_time`.

Also, seems like the `system boot` entry does not exist on at least RHEL or Mac OS in whatever utmp file `getutxent` reads.